File: app/src/androidTest/java/org/andstatus/todoagenda/TomorrowsTasksInTodaysFilterTest.java
package org.andstatus.todoagenda;

import static org.junit.Assert.assertEquals;

import org.andstatus.todoagenda.provider.QueryResultsStorage;
import org.andstatus.todoagenda.widget.WidgetEntryPosition;
import org.junit.Test;

/**
 * @author yvolk@yurivolkov.com
 * https://github.com/andstatus/todoagenda/issues/102
 */
public class TomorrowsTasksInTodaysFilterTest extends BaseWidgetTest {
    @Test
    public void testIssue102() {
        final String method = "testIssue102";
        QueryResultsStorage inputs = provider.loadResultsAndSettings(
            org.andstatus.todoagenda.test.R.raw.tomorrows_tasks_today);
        provider.addResults(inputs);

        playResults(method);
        assertEquals("Number of entries, " + getFactory().getWidgetEntries(), 1, getFactory().getWidgetEntries().size());
        assertEquals("No events", WidgetEntryPosition.LIST_FOOTER, getFactory().getWidgetEntries().get(0).entryPosition);
    }
}

File: app/src/androidTest/kotlin/org/andstatus/todoagenda/TomorrowsTasksInTodaysFilterTest.kt
package org.andstatus.todoagenda

import org.andstatus.todoagenda.prefs.InstanceSettings
import org.andstatus.todoagenda.widget.LastEntryType
import org.junit.Test

/**
 * @author yvolk@yurivolkov.com
 * https://github.com/andstatus/todoagenda/issues/102
 */
class TomorrowsTasksInTodaysFilterTest : BaseWidgetTest() {
    @Test
    fun testIssue102() {
        val method = "testIssue102"
        provider.loadResultsAndSettings(org.andstatus.todoagenda.test.R.raw.tomorrows_tasks_one_week)
        playResults(method)
        assertLastEntry(12, LastEntryType.END_OF_LIST)
        provider.settings =
            settings.copy(
                eventRange = InstanceSettings.EVENT_RANGE_TODAY,
            )
        playResults(method)
        assertLastEntry(0, LastEntryType.NO_EVENTS)
        provider.settings =
            settings.copy(
                eventRange = InstanceSettings.EVENT_RANGE_TODAY_AND_TOMORROW,
            )
        playResults(method)
        assertLastEntry(9, LastEntryType.END_OF_LIST)
    }
}
